wagtail/core/fields.py

```diff
@@ -51,13 +51,31 @@ def __set__(self, obj, value):
 
 class StreamField(models.Field):
     def __init__(self, block_types, **kwargs):
+
+        # extract kwargs that are to be passed on to the block, not handled by super
+        block_opts = {}
+        for arg in ['min_num', 'max_num', 'block_counts']:
+            if arg in kwargs:
+                block_opts[arg] = kwargs.pop(arg)
+
+        # for a top-level block, the 'blank' kwarg (defaulting to False) always overrides the
+        # block's own 'required' meta attribute, even if not passed explicitly; this ensures
+        # that the field and block have consistent definitions
+        block_opts['required'] = not kwargs.get('blank', False)
+
         super().__init__(**kwargs)
+
         if isinstance(block_types, Block):
+            # use the passed block as the top-level block
             self.stream_block = block_types
         elif isinstance(block_types, type):
-            self.stream_block = block_types(required=not self.blank)
+            # block passed as a class - instantiate it
+            self.stream_block = block_types()
         else:
-            self.stream_block = StreamBlock(block_types, required=not self.blank)
+            # construct a top-level StreamBlock from the list of block types
+            self.stream_block = StreamBlock(block_types)
+
+        self.stream_block.set_meta_options(block_opts)
```

Comment:
What if the options were specificed that aren't supported by the root block?

Maybe we should pass them as kwargs here so a ``TypeError`` is raised?

Reply:
Passing as kwargs is messy because we need to distinguish between `max_num=None` (allow unlimited blocks, overriding any maximum set at the class level) and not passing `max_num` (leave the setting unchanged from what the class definition says). Best to cross that bridge when we start supporting other top-level block types besides StreamBlock, I think...

Reply:
Actually, scratch that - I now see that `StreamBlock.set_meta_options` is really just duplicating the information in `MUTABLE_META_ATTRIBUTES`, so we can make the base `set_meta_options` method a bit smarter and fix this and the code comment issue at the same time :-)